File: smartstore_backend/src/main/java/com/ssafy/smartstore/model/dao/FavoriteDao.java
package com.ssafy.smartstore.model.dao;

import com.ssafy.smartstore.model.dto.Favorite;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Map;

@Component
public interface FavoriteDao {
    int insert(Favorite favorite);

    List<Map<String, Object>> select(String userId);

    int delete(String id);
}

File: Smartstore_Network-starter/app/src/main/java/com/ssafy/smartstore/database/FavoriteDao.kt
package com.ssafy.smartstore.database

import androidx.room.*

@Dao
interface FavoriteDao {

    @Query("SELECT * FROM favorite")
    suspend fun getFavorites() : MutableList<FavoriteDto>

    @Query("SELECT * FROM favorite WHERE productId = (:id)")
    suspend fun getFavorite(id: Int) : FavoriteDto

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertFavorite(dto : FavoriteDto)

    @Update
    suspend fun updateFavorite(dto : FavoriteDto)

    @Query("DELETE FROM favorite WHERE productId = (:id)")
    suspend fun deleteFavorite(id:Int)

}


